src/contracts/tokens/TokenDistributor.sol

```diff
@@ -33,53 +30,31 @@ contract TokenDistributor is Authorizable, ITokenDistributor {
   mapping(address _user => bool _hasClaimed) public claimed;
 
   /**
-   * @param  _token Address of the ERC20 token to be distributed
-   * @param  _tokenDistributorParams TokenDistributor valid parameters struct
+   * @param _token Address of the ERC20 token to be distributed
+   * @param _root Bytes32 representation of the merkle root
+   * @param _totalClaimable Total amount of tokens to be distributed
+   * @param _claimPeriodStart Timestamp when the claim period starts
+   * @param _claimPeriodEnd Timestamp when the claim period ends
    */
-  constructor(address _token, TokenDistributorParams memory _tokenDistributorParams) Authorizable(msg.sender) {
-    token = IProtocolToken(_token.assertHasCode());
-    root = _tokenDistributorParams.root;
-    totalClaimable = _tokenDistributorParams.totalClaimable.assertNonNull();
-    claimPeriodStart = _tokenDistributorParams.claimPeriodStart.assertGt(block.timestamp);
-    claimPeriodEnd = _tokenDistributorParams.claimPeriodEnd.assertGt(claimPeriodStart);
+  constructor(
+    address _token,
+    bytes32 _root,
+    uint256 _totalClaimable,
+    uint256 _claimPeriodStart,
+    uint256 _claimPeriodEnd
+  ) Authorizable(msg.sender) {
+    // token = _token.assertHasCode();
```

Comment:
👀 !!!

Reply:
ah, you're skipping declaring `token` bc in one is ERC20Votes and in the other one just ERC20, i'd store it as address and cast it to the correct interface when needed